hibernate-core/src/main/java/org/hibernate/internal/FetchingScrollableResultsImpl.java

```diff
@@ -96,15 +106,13 @@ public boolean next() {
 
 	@Override
 	public boolean previous() {
-		if ( currentPosition <= 1 ) {
-			currentPosition = 0;
-			currentRow = null;
+		if ( beforeFirst || isResultSetEmpty() ) {
 			return false;
 		}
-
-		if ( getRowProcessingState().isFirst() ) {
-			// don't even bother trying to read any further
-			currentRow = null;
+		else if ( currentPosition == 1 ) {
+			beforeFirst();
+			afterLast = false;
```

Comment:
Minor thing, this seems unnecessary.

```suggestion
```

Reply:
You're right, I added it while troubleshooting a test that was failing and forgot to remove it, thanks.